.github/workflows/rust.yml

```diff
@@ -0,0 +1,22 @@
+name: Rust
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+env:
```

Comment:
Do we actually need this for the code to run?

Reply:
i think it could be cool to have PRs checked. So yep, otherwise only master will have CI enabled.